extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxBuilderDialog.js

```diff
@@ -0,0 +1,107 @@
+/* globals require */
+
+/*
+ * VisualEditor user interface WikiaInfobooxBuilderDialog class.
+ */
+
+/**
+ * Dialog for inserting portable infobox templates.
+ *
+ * @class
+ * @extends ve.ui.Dialog
+ *
+ * @constructor
+ * @param {Object} [config] Configuration options
+ */
+ve.ui.WikiaInfoboxBuilderDialog = function VeUiWikiaInfoboxBuilderDialog(config) {
+	// Parent constructor
+	ve.ui.WikiaInfoboxBuilderDialog.super.call(this, config);
+
+	$(window).on('infoboxBuilderReturnToVE', function (event, isOnPublish) {
```

Comment:
is this the recommended way of interacting between different OOJS components? Do we have anything about it in OOJS documentation ?